contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,132 @@
 pragma solidity ^0.8.5;
 
+/**
+ * @notice The Gitcoin GrantRegistry contract keeps track of all grants that have been created.
+ * It is designed to be a singleton, i.e. there is only one instance of a GrantRegistry which
+ * tracks all grants. It behaves as follows:
+ *   - Anyone can create a grant by calling `createGrant`
+ *   - A grant's `owner` can edit their grant using the `updateGrant` family of method
+ *   - The `getAllGrants` and `getGrants` view methods are used to fetch grant data
+ */
 contract GrantRegistry {
-  // Grant ID - The hash of the grants meetadata
-  bytes32 public grantRegistryId;
-  // Metadata Pointer
-  string public metaPtr;
-
-  // Owner
-  address owner;
-
-  modifier onlyOwner() {
-    require(msg.sender == owner);
-    _;
-  }
-
-  // TODO: Only allow the owner (which is the curator) address to configure the contract
-  // while its in config mode, then switch to active. Once active, the owner can only
-  // close the registry (have a conversation around this approach)
-  constructor(
-    address _owner,
-    bytes32 _grantRegistryId,
-    string memory _metaPtr
-  ) {
-    owner = _owner;
-    //state = State.CONFIG;
-    grantRegistryId = _grantRegistryId;
-    metaPtr = _metaPtr;
-  }
-
-  // State
-  enum State {
-    CONFIG,
-    ACTIVE,
-    CLOSED
-  }
-  //State public state;
-
-  // Grant Pointer Object
-  // Mapping from GrantID to set of Owner Address, Payout Address, Metadata Pointer, Replaces Grant (optional)
-  // TODO: Implement safemath for avoiding overflows with ints
-  uint256 public grantCount = 0;
-  mapping(bytes32 => Grant) public grants;
-  bytes32[] public grantIds;
+  // --- Data ---
+  /// @notice Number of grants stored in this registry
+  uint96 public grantCount;
 
+  /// @notice Grant object
   struct Grant {
-    bytes32 id;
-    address owner;
-    address payout;
-    string metaPtr;
-    State state;
-    bytes32 replaces;
+    uint96 id; // grant ID, as uint96 to pack into same slot as owner (this implies a max of 2^96-1 = 7.9e28 grants)
+    address owner; // grant owner (has permissions to modify grant information)
+    address payee; // address that receives funds donated to this grant
+    string metaPtr; // URL pointing to grant metadata (for off-chain use)
   }
```

Comment:
Is it overkill to have a status for Grant ?
Status to indicate -> open for funding / closed for funding

Reply:
Hmm, I don't think we need it, since that can be part of the JSON schema that will live off-chain at the `metaPtr` URL. But curious to hear what @apbendi or @phutchins think

Reply:
If metadata is unavailable and the status lives in the metadata, how do we handle displaying a grant? This type of situation makes me think that it should live in the grant however in this early stage it prob doesn't mater as much. I'd. say that we leave it in the grant for now to make it easier for filter and display and then move it to the metadata if we determine that the cost is too high.

Reply:
One idea @apbendi mentioned is we could also just set the payee address to the zero address to indicate a grant is not available for funding, that way we don't need another parameter here. OTOH, we can pack the status into the `payee` slot so it won't take up any additional storage space/gas

Reply:
Ah yes that would be perfect @mds1